File: java/BaseAndroidViewModel.java
package dita.dev.myportal.ui.base.v2;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.support.annotation.NonNull;

public class BaseAndroidViewModel extends AndroidViewModel {
    private final SnackbarMessage snackbarMessage = new SnackbarMessage();

    private final Finish finishEvent = new Finish();

    public BaseAndroidViewModel(@NonNull Application application) {
        super(application);
    }

    public SnackbarMessage getSnackbarMessage() {
        return snackbarMessage;
    }

    public Finish getFinishEvent() {
        return finishEvent;
    }

    protected void showSnackbar(String message) {
        snackbarMessage.setValue(message);
    }

    protected void showSnackbarBackground(String message) {
        snackbarMessage.postValue(message);
    }

    protected void finish(Class clazz) {
        finishEvent.setValue(clazz);
    }

}

File: kotlin/BaseAndroidViewModel.kt
import android.app.Application
import android.arch.lifecycle.AndroidViewModel

open class BaseAndroidViewModel constructor(application: Application) : AndroidViewModel(application) {
    open val snackbarMessage = SnackbarMessage()
    open val finishEvent = Finish()

    fun showSnackbar(message: String) {
        snackbarMessage.value = message
    }

    fun showSnackbarBackground(message: String) {
        snackbarMessage.postValue(message)
    }

    fun finish(clazz: Class<Any>) {
        finishEvent.value = clazz
    }
}
